Extend social auth saml team attr with admin and superuser option

CoolZeroNL opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
I Extended the SOCIAL_AUTH_SAML_TEAM_ATTR so that you can assign admins and/or superuser trough SAML.

<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - SSO Authentication SAML (SOCIAL_AUTH_SAML_TEAM_ATTR)
 - awx/sso/pipeline.py
 - awx/sso/fields.py
- awx/docs/auth/saml.md

##### AWX VERSION
<!--- Paste verbatim output from `make VERSION` between quotes below -->
```
awx: 9.2.0
```
##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

_AttributeStatement:_
 * groupIds                                                           = awx-admins
 * groupIds                                                           = awx-superusers
 * groupIds                                                           = awx-members

```
<saml:AttributeStatement>
            <saml:Attribute Name="groupIds" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:basic">
                <saml:AttributeValue xsi:type="xs:string">allUsers</saml:AttributeValue>
                <saml:AttributeValue xsi:type="xs:string">awx-admins</saml:AttributeValue>
                <saml:AttributeValue xsi:type="xs:string">awx-superusers</saml:AttributeValue>
                <saml:AttributeValue xsi:type="xs:string">awx-members</saml:AttributeValue>
                <saml:AttributeValue xsi:type="xs:string">awx-auditors</saml:AttributeValue>
            </saml:Attribute>
</saml:AttributeStatement>
```

_SAML TEAM ATTRIBUTE MAPPING:_
```
{
 "team_org_map": [
  {
   "organization": "awx-members",
   "team": "awx-members"
  },
  {
   "admins": true,
   "organization": "awx-admins",
   "team": "awx-admins"
  },
  {
   "auditors": true,
   "organization": "awx-auditors",
   "team": "awx-auditors"
  },
  {
   "superusers": true,
   "organization": "awx-superusers",
   "team": "awx-superusers"
  }
 ],
 "saml_attr": "groupIds",
 "remove": true,
 "admins_remove": true,
 "auditors_remove": true,
 "superusers_remove": true
}
```

CoolZeroNL committed:
Adding Admin/Superuser Logic into SAML_TEAM_ATTR

CoolZeroNL committed:
Adding Fields for admin and Superuser

Update SAMLTeamAttrTeamOrgMapField with:
- admins
- superusers

Update SAMLTeamAttrField with:
- admins_remove
- superusers_remove

CoolZeroNL committed:
Extending SAML Team Attribute Mapping

Extending Example with admins, and superusers.

ryanpetrello commented:
@wenottingham do we have an issue where we're tracking this enhancement?

wenottingham commented:
This looks like it would be for https://github.com/ansible/awx/issues/5303.

It should be updated to handle auditors, though.

CoolZeroNL committed:
Adding Auditors Logic into SAML_TEAM_ATTR

CoolZeroNL committed:
Adding Auditors fields

adding to SAMLTeamAttrTeamOrgMapField: 
- auditors

adding to SAMLTeamAttrField:
-  auditors_remove

CoolZeroNL committed:
Add Auditors to Team Attribute Mapping example

CoolZeroNL commented:
> It should be updated to handle auditors, though.

@wenottingham --> Added the auditors.

CoolZeroNL commented:
recheck

CoolZeroNL commented:
@wenottingham @ryanpetrello Is there anything more that is needed in this PR to get this reviewed/merged ?

ryanpetrello commented:
@CoolZeroNL 

More hours in the day 😄.

It's on our list to look this over, but we just haven't found time for it yet (setting up regression testing is the actual time consuming part).

wenottingham commented:
Looking again, this isn't quite right - you wouldn't map global sysadmins and auditors as part of SOCIAL_AUTH_SAML_TEAM_ATTR. Those are global user flags, unrelated to teams.

nicolas-g commented:
hi any updates on this ? we have encounter an issue and it seems this PR will fix it ...

shanemcd commented:
Closing this because the feedback provided was never addressed.
